### DISCUSSION TITLE
Windowing streams on Vector?

### DISCUSSION THREAD

**Discussion by igr001-galactica** [vectordotdev/vector#13055] [Q&A] [answered]
I have worked with Kafka Streams in the past where I have been able to define a rolling time window in which to do transformations/filters/aggregations. Is something similar possible on Vector?
I would like to define a rolling time window, say 1 minute and perform some deduplications.

**Comment by jszwedko** [comment, marked as answer]
Hi @igr001-galactica !

The `reduce` transform does operate sort of like that, to collapse multiple logs into one based on conditions (like time), but generally Vector doesn't have great support for rolling window aggregation like that yet. It is something we are thinking about though. For example: https://github.com/vectordotdev/vector/issues/1329

**Reply by JeanMertz** [reply]
Closing in favour of the linked tracking issue at https://github.com/vectordotdev/vector/issues/1329.
